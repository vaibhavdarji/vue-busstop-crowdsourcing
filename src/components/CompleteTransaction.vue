<template>
  <v-card>
    <v-container grid-list-xl fluid>
      <v-layout row>
         <v-flex xs12>
           <h2>
              Thank you for supporting our Move {{selectedBusStop.name}} with your donation of
              <v-icon>attach_money</v-icon>
              {{userDonation.amount}}.
           </h2>
         </v-flex>  
       </v-layout>
       <v-card-actions>
          <v-spacer></v-spacer>
          <custom-button @onClick="onComplete">
            Done
          </custom-button>
      </v-card-actions>
    </v-container>
  </v-card>
</template>

<script>
import { mapGetters, mapActions } from 'vuex';
import CustomButton from '@/components/Button.vue';
import {
  RESET_TRANSACTION
} from '@/actions/types';

export default {
  name: 'CompleteTransaction',
  data () {
    return {};
  },

  components: {
    CustomButton
  },

  computed: {
    ...mapGetters([
      'selectedBusStop',
      'userDonation'
    ])
  },

  methods: {
    ...mapActions({
      'resetTransaction': RESET_TRANSACTION
    }),
    onComplete () {
      this.resetTransaction();
      this.$router.push('/');
    }
  }
}
</script>

