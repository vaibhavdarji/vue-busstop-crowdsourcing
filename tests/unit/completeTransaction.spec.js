import { shallowMount, mount } from "@vue/test-utils";
import CompleteTransaction from '@/components/CompleteTransaction.vue';

import Vue from "vue";
import Vuex from "vuex";
import Vuetify from "vuetify";
import Constants from '@/constants';

Vue.use(Vuex);
Vue.use(Vuetify);

const busStop =  {stopId: 1, lat: 33.760262, lng: -84.384706, donationsRaisedInDollars: 0, name: 'Hertz at Portman Blvd'};

describe("CompleteTransaction.vue", () => {
  let getters, store;
  beforeEach(() => {
    getters = {
      selectedBusStop : () => busStop,
      userDonation: () => ({
        ...Constants.USER_INFO,
        ...{
          amount: 700
        }
      })
    };
    store = new Vuex.Store({
      getters
    });
  });

  it("should match snapshot", () => {
    const wrapper = shallowMount(CompleteTransaction, {
      store,
      Vue
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});