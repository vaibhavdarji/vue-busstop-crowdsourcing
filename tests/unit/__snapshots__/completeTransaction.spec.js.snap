// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompleteTransaction.vue should match snapshot 1`] = `
<v-card-stub
  tag="div"
>
  <v-container-stub
    fluid=""
    grid-list-xl=""
    tag="div"
  >
    <v-layout-stub
      row=""
      tag="div"
    >
      <v-flex-stub
        tag="div"
        xs12=""
      >
        <h2>
          
            Thank you for supporting our Move Hertz at Portman Blvd with your donation of
            
          <v-icon-stub>
            attach_money
          </v-icon-stub>
          
            700.
         
        </h2>
      </v-flex-stub>
    </v-layout-stub>
     
    <v-card-actions-stub>
      <v-spacer-stub />
       
      <custom-button-stub
        btntype="button"
      >
        
          Done
        
      </custom-button-stub>
    </v-card-actions-stub>
  </v-container-stub>
</v-card-stub>
`;
